src/api/parser/src/lib/shutdown.js

```diff
@@ -0,0 +1,66 @@
+const { promisify } = require('util');
```

Comment:
We might need to integrate this into how Satellite does its graceful shutdown code.  We need to investigate later.